### ISSUE TITLE
Cannot read property 'product' of null

### ISSUE THREAD

**Issue by stephensilber** [ndimatteo/HULL#47]
I'm seeing my Shopify products come through into Sanity via the webhook but whenever I add a product to a page and click it, I am met with the following error when navigating to the URL (`/products/<slug>`)
```
Server Error
TypeError: Cannot read property 'product' of null

This error happened while generating the page. Any console logs will be displayed in the terminal window.
Source
pages/products/[slug].js (60:46) @ Product

  58 |   console.log(data)
  59 |   // set our Product state
> 60 |   const [product, setProduct] = useState(page.product)
     |                                              ^
  61 | 
  62 |   // find the default variant for this product by matching against the first product option
  63 |   const defaultVariant = page.product.variants?.find((v) => {
```

Is there something special I need to do for a Shopify product to have a "detail" page?

**Comment by ndimatteo** [comment]
Hey there @stephensilber is this on the latest version of HULL or an older version?

It sounds like it's not reading the `product` object that is sent through so it's unable to generate the page.

Very hard to tell from this what the issue is.

**Comment by stephensilber** [comment]
I think I generated this project ~3 weeks ago if that helps give an indication of what version it's on. It actually looks like the `page` object comes back null. Here's the `data` property printed out from the file above `(/prodcuts/[slug].js)`

```
{
  page: null,
  site: {
    cart: {
      message: 'Free shipping on orders over $666',
      storeURL: 'https://shop.hull.dev'
    },
    cookieConsent: {
      enabled: true,
      link: [Object],
      message: 'We use cookies to personalize and deliver appropriate content.\n' +
        'By clicking "Accept" you agree to our terms.'
    },
    footer: { blocks: [Array] },
    header: {
      menuDesktopLeft: [Object],
      menuDesktopRight: [Object],
      menuMobilePrimary: [Object],
      menuMobileSecondary: [Object],
      promo: [Object]
    },
    productCounts: [ [Object], [Object], [Object], [Object], [Object], [Object] ],
    rootDomain: 'https://soho-ski-club.vercel.app',
    seo: {
      metaDesc: 'A nature club for city dwellers. Discover premium basics for the adventure obsessed.',
      metaTitle: 'Soho Ski Club',
      shareDesc: 'A nature club for city dwellers. Discover premium basics for the adventure obsessed.',
      shareGraphic: [Object],
      shareTitle: 'Soho Ski Club',
      siteTitle: 'Soho Ski Club'
    }
  }
}
```

**Comment by ndimatteo** [comment]
Thanks @stephensilber, so I pushed a major update on **[sept 19](https://github.com/ndimatteo/HULL/pull/40)**, so if you pulled a copy before then you might be on an out-dated version I'm afraid. If so, would it be possible for you to rebuild with the latest version?

I'm not sure why your `page` object would be `null`, but that data is generated from here for product pages: https://github.com/ndimatteo/HULL/blob/802f5309c7f060cb7187f606f2907aadf2b33ee3/data/index.js#L60-L79

That GROQ should be grabbing your product, which would give you data for the `page` object. Maybe that will help uncover what's missing on your end?

If you're on the latest, and still not able to figure it out, feel free to add me to your repo and sanity project so I can take a look locally.

**Comment by stephensilber** [comment]
is there instructions on how I can rebase off of our latest changes without causing a bunch of issues on my end? I really appreciate the quick responses!!

**Comment by ndimatteo** [comment]
@stephensilber no problem!

Unfortunately, I don't have any specific instructions on how to do that but perhaps this would help:

https://stackoverflow.com/questions/56577184/github-pull-changes-from-a-template-repository

**Comment by stephensilber** [comment]
Thanks for the link! I figured out that my issue was related to the `isDraft` flag in the above `getProduct` query (my client had uploaded all of their products but had not set them as active). I did try to pull in the latest changes via that stack overflow post but I was looking at ~35-40 files with merge conflicts so quickly gave up on that. Gonna mark this as closed...thanks again for the quick responses.
